### PR TITLE
Possibility to specify complex data for input's value (v2 backport)

### LINKED ISSUE #148: input: Could not specify localized input as value from BEMJSON
```
{ block: 'input', mods: { size: 's' }, name: 'my-name', value: {
    block: 'i-bem', elem: 'i18n',
    keyset: 'my-keyset', key: 'my-key'
}, content: { elem: 'input' } }
```

### PR DESCRIPTION
**Pull request by narqo**
\cc @veged 

Could you review BEMHTML template, 'cause I still got few experience with new JS-syntax?

Fix #148 (v2)

### CODE REVIEW

File: common.blocks/input/input.bemhtml [hunk 1]
```diff
@@ -1,11 +1,27 @@
 block('input')(
     tag()('span'),
+
     js()(true),
+
     def()(function() {
         var ctx = this.ctx;
         ctx.id = ctx.id || this.generateId();
+        ctx.val = apply('value', { ctx: ctx.val });
```
Comment (veged) [inline common.blocks/input/input.bemhtml]: I suggest simplify separate mode and applying process to:

``` js
var val = ctx.val;
ctx.val = this._.isSimple(val) ? val : BEMHTML.apply(val);
```

if we really need to extend something with calculating value, we can introduce separate templates mode later

Comment (narqo) [inline common.blocks/input/input.bemhtml]: «Ох тыж ёжик...» o_O``

Comment (veged) [inline common.blocks/input/input.bemhtml]: ?

**Comment by narqo** [comment]
See #195

**Pull request closed without merging**